framework/deproxy_manager.py

```diff
@@ -0,0 +1,61 @@
+import Queue
+import threading
+import asyncore
+import select
+import time
+
+from helpers import stateful, tf_cfg
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2018 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+def finish_all_deproxy():
+    asyncore.close_all()
+
+def run_deproxy_server(deproxy, exit_event, polling_lock):
+    tf_cfg.dbg(3, "Running deproxy server manager")
+
+    if hasattr(select, 'poll'):
+        poll_fun = asyncore.poll2
+    else:
+        poll_fun = asyncore.poll
+    while not exit_event.is_set():
+        polling_lock.acquire()
+        poll_fun()
+        polling_lock.release()
+
+    tf_cfg.dbg(3, "Stopped deproxy manager")
+
+class DeproxyManager(stateful.Stateful):
+
```

Comment:
Please write a good comment for the class - it's not clear why do we need the class and what it's responsible for. I don't understand why it's not in `helpers/deproxy.py`? Why `Deproxy` can/should not be responsible for, for example, adding a new server (`add_server()`)?

Basically the same questions with `deproxy_client.py` and `deproxy_server.py`. It seems they implement some logic which should be in `deproxy.py`... Or deproxy is large enough to be placed in separate directory, out of `helpers`, and have several `.py` files. The architecture isn't clear.

If the reason for the "strange" organisation is that you're trying to minimize required changes, i.e. https://github.com/tempesta-tech/tempesta-test/pull/5#issuecomment-384681562 , then we should allocate more time for the task. If you want to keep old test working then, we can back them to `tempesta_fw` while `tempesta-test` will just copy necessary and useful classes from the old framework and do the things properly. We can maintain/fix old test for some time in `tempesta_fw` until we finally move to the new framework.

The target for #878 is to get robust and clear framework for the tests. The target is likely to be failed if you has to couple the old architecture with the new one.